lib/guard/puma.rb

```diff
@@ -19,40 +19,56 @@ def self.default_env
       :start_on_start => true,
       :force_run => false,
       :timeout => 20,
-      :debugger => false
+      :debugger => false,
+      :notifications => %i[restarting restarted not_restarted stopped]
     }
 
     def initialize(options = {})
       super
       @options = DEFAULT_OPTIONS.merge(options)
+      @options[:port] = nil if @options.key?(:config)
       @runner = ::Guard::PumaRunner.new(@options)
     end
 
     def start
       server = options[:server] ? "#{options[:server]} and " : ""
-      UI.info "Guard::Puma will now start your app on port #{options[:port]} using #{server}#{options[:environment]} environment."
+      UI.info "Guard::Puma will now start your app#{port_text} using #{server}#{options[:environment]} environment."
       runner.start if options[:start_on_start]
     end
 
     def reload
       UI.info "Restarting Puma..."
-      Notifier.notify("Puma restarting on port #{options[:port]} in #{options[:environment]} environment...", :title => "Restarting Puma...", :image => :pending)
+      if options[:notifications].include?(:restarting)
+        Notifier.notify("Puma restarting#{port_text} in #{options[:environment]} environment...", :title => "Restarting Puma...", :image => :pending)
+      end
       if runner.restart
         UI.info "Puma restarted"
-        Notifier.notify("Puma restarted on port #{options[:port]}.", :title => "Puma restarted!", :image => :success)
+        if options[:notifications].include?(:restarted)
+          Notifier.notify("Puma restarted#{port_text}.", :title => "Puma restarted!", :image => :success)
+        end
       else
         UI.info "Puma NOT restarted, check your log files."
-        Notifier.notify("Puma NOT restarted, check your log files.", :title => "Puma NOT restarted!", :image => :failed)
+        if options[:notifications].include?(:not_restarted)
+          Notifier.notify("Puma NOT restarted, check your log files.", :title => "Puma NOT restarted!", :image => :failed)
+        end
       end
     end
 
     def stop
-      Notifier.notify("Until next time...", :title => "Puma shutting down.", :image => :pending)
+      if options[:notifications].include?(:stopped)
+        Notifier.notify("Until next time...", :title => "Puma shutting down.", :image => :pending)
+      end
       runner.halt
     end
 
-    def run_on_change(paths)
+    def run_on_changes(paths)
       reload
     end
+
+    private
+
+    def port_text
+      options[:port] ? " on port #{options[:port]}" : ""
```

Comment:
I'm not a fan of `? :` like this. What do you think about

```ruby
if port = options[:port]
  " on port #{port}"
end
```
or even
```ruby
return " on port #{options[:port]}" if options[:port]"
```
and we allow `nil` to be returned instead of `""`?

Reply:
Yes, thanks, the last one is the best (but `return` doesn't necessary :grimacing:)